docs/OnlineTutorial/Lemmas.md

```diff
@@ -786,45 +774,45 @@ we can temporarily put preconditions on our lemma as follows:
 ``` {.editonly}
 lemma DisproofLemma(p: seq<Node>, subgraph: set<Node>,
                     root: Node, goal: Node, graph: set<Node>)
-   requires closed(subgraph) && closed(graph) && subgraph <= graph
-   requires root in subgraph && goal in graph - subgraph
-   requires |p| < 2 || p[0] != root || p[|p|-1] != goal
-   ensures !pathSpecific(p, root, goal, graph)
+  requires closed(subgraph) && closed(graph) && subgraph <= graph
+  requires root in subgraph && goal in graph - subgraph
+  requires |p| < 2 || p[0] != root || p[|p|-1] != goal
+  ensures !pathSpecific(p, root, goal, graph)
 {
 }
 class Node
 {
-   var next: seq<Node>
+  var next: seq<Node>
 }
 predicate pathSpecific(p: seq<Node>, start: Node, end: Node, graph: set<Node>)
-   requires closed(graph)
-   reads graph
+  requires closed(graph)
+  reads graph
 {
-   0 < |p| && // path is nonempty
-   start == p[0] && end == p[|p|-1] && // it starts and ends correctly
-   path(p, graph) // and it is a valid path
+  0 < |p| && // path is nonempty
+    start == p[0] && end == p[|p|-1] && // it starts and ends correctly
+    path(p, graph) // and it is a valid path
 }
 predicate path(p: seq<Node>, graph: set<Node>)
-   requires closed(graph) && 0 < |p|
-   reads graph
+  requires closed(graph) && 0 < |p|
+  reads graph
 {
-   p[0] in graph &&
-   (|p| > 1 ==> p[1] in p[0].next && // the first link is valid, if it exists
-      path(p[1..], graph)) // and the rest of the sequence is a valid
+  p[0] in graph &&
+    (|p| > 1 ==> p[1] in p[0].next && // the first link is valid, if it exists
+    path(p[1..], graph)) // and the rest of the sequence is a valid
```

Comment:
```suggestion
  p[0] in graph &&
    (|p| > 1 ==> p[1] in p[0].next && // the first link is valid, if it exists
     path(p[1..], graph)) // and the rest of the sequence is a valid
```
Is it me or it's the fourth time I'm seeing the same code?

Reply:
It is a tutorial ... the same example keeps being enhanced